src/mixxx.cpp

```diff
@@ -598,6 +596,11 @@ void MixxxMainWindow::initializeWindow() {
     Pal.setColor(QPalette::Background, MenuBarBackground);
     m_pMenuBar->setPalette(Pal);
 
+    // restore window size/position
+    QSettings settings("mixxxdj", "mixxx");
```

Comment:
> spreading state across more files means more maintenance burden and more complexity

Can you please elaborate? What maintenance burden and complexity? IMO it's the opposite way around; writing wrapper code to work with the .cfg file adds complexity and maintenance burden. Letting QSettings take care of miscellaneous widget states works for free without cluttering the cfg files or adding any maintenance burden.